controllers/schedules_controller.py

```diff
@@ -10,24 +11,32 @@ class SchedulesController(Resource):
     def __init__(self):
         self.parser = reqparse.RequestParser()
         self.interface = FirebaseInterface()
-        self.schedule = Schedule()
 
     def post(self):
         req = request.get_json()
 
         try:
-            office_id = req["id_oficina"]
-            office = self.interface.getData("offices", office_id)
+            self.validateRequest(req)
 
-            if office is None:
-                raise Exception("Oficina não encontrada")
+            schedule = Schedule(req)
 
-            self.schedule.buildObject(req)
-            self.schedule.validateFields(office["agenda"])
-            self.schedule.setId()
+            if schedule.id is None:
```

Comment:
Você chegou a testar se a o request for enviado com um json vazio, não pode dar um NullPointer nesse "schedule.id"? Ou to viajando kkkk

Reply:
Essa validação eu faço no objeto Schedule, coloco None caso o Json não possua o campo id senão preencho com o que veio. Assim, dentro do post e do put faço as validações que preciso desse atributo